### ISSUE TITLE
New tree-sitter parsing added to 1.32.0 broke syntax highlighting for CSS within HTML

### ISSUE THREAD

**Issue by arodic** [atom/atom#18327]
### Description

It appears that new three-sitter parsing added to 1.32.0 broke nested language syntax highlighting in html template literals in javascript and possibly other languages.

### Steps to Reproduce

1. Create a javascript file with following code:

```javascript
var code = html`
  <style>
    div {
      display: flex;
    }
  </style>
`;
```

**Expected behavior:** CSS code should be highlighted just like in this example above.

**Actual behavior:** CSS code is displayed in single color like plain text.

**Reproduces how often:** Always

### Versions

1.32.0

**Comment by arodic** [comment]
My best guess is that #17879 broke the nested syntax highlighting.

**Comment by arodic** [comment]
Can confirm, `useTreeSitterParsers: false` results with correct CSS syntax highlighting.

**Comment by pjrebsch** [comment]
@Ben3eeE You mention a known issue with CSS parsing, but I've experienced issues with syntax highlighting in ERB files since updating to 1.32.0. I don't think I can reproduce intentionally, but it did occur to me several times today where syntax highlighting will break during edits and eventually Atom will freeze with high CPU usage for about 10 seconds at a time. It doesn't seem to resolve itself unless I close the file and reopen it.

**Comment by hermidalc** [comment]
Adding Python syntax highlighting is also broken in in 1.32, worked fine in 1.31.2.

**Comment by Ben3eeE** [comment]
@hermidalc If you open a new issue on the https://github.com/atom/language-python repository and fill out the issue template we can take a look.

**Comment by dwightware4** [comment]
This also appears to have broken JSX highlighting for React. Setting `useTreeSitterParsers: false` makes it work fine again.

**Comment by Ben3eeE** [comment]
@dwightware4 This issue is about CSS. JSX highlighting has been fixed, please see https://github.com/atom/atom/issues/18307#issuecomment-433236880

If you have a specific issue that is not mentioned in https://github.com/atom/atom/issues/18307 then please open a new issue with details and we can take a look.

**Comment by maxbrunsfeld** [comment]
Yup, just to add my 2 cents: sorry for shipping this without that feature. We're going to add it back as soon as we can.

**Comment by Dominik1799** [comment]
Is there any way to fix this?

**Comment by Arcanemagus** [comment]
@Dominik1799 see the previous comment here: https://github.com/atom/atom/issues/18327#issuecomment-433034205

**Comment by someburner** [comment]
> and possibly other languages.

Can confirm- unticking `useTreeSitterParsers` in Core settings fixes broken syntax highlighting in C code where `#ifdef`s are present.

**Comment by maxbrunsfeld** [comment]
@someburner Can you open a separate issue on the [atom/language-c](https://github.com/atom/language-c) repo describing the problem you're seeing with `#ifdef`?

**Comment by someburner** [comment]
@maxbrunsfeld 

Oops- it's already there, sorry. Wasn't aware it needed to be fixed for each language.

**Comment by maxbrunsfeld** [comment]
This has been fixed via https://github.com/atom/language-css/pull/156. Sorry for the breakage, everyone. The fix will go out with Atom 1.32.1.

If you re-enable Tree-sitter and find other highlighting missing, please open a new issue on the relevant language package (e.g. [language-javascript](https://github.com/atom/language-javascript)), if an issue doesn't already exist.

**Comment by Yurickh** [comment]
Even though highlight is fixed by now, I don't get css autocomplete suggestions within the css. Is this expected behaviour? If it isn't, which project is responsible for this feature? Is it autocomplete-plus, language-javascript or language-css?
